### PR TITLE
[CM T3] Kotlin binding refinement: null values, restructure, find methods

### PR DESCRIPTION
**Pull request by mandoway**
Fixes according to #4368 
Also includes new find functions on KeySets

## Basics

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [x] The PR is rebased with current master.

## Checklist

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned [every code](/.reuse/dep5) not directly written by me in [reuse syntax](https://reuse.software/)

## Review

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

- [ ] Add the "work in progress" label if you do not want the PR to be reviewed yet.
- [ ] Add the "ready to merge" label **if the basics are fulfilled** and no further pushes are planned by you.

### CODE REVIEW

File: src/bindings/jna/libelektra-kotlin/src/main/kotlin/org/libelektra/keySetExt/Util.kt [hunk 1]
```diff
@@ -4,11 +4,14 @@ import org.libelektra.Key
...
- * @param keys will be added to the new KeySet
- * @return KeySet containing all given keys
+ * @param keyName name of the key to lookup in the KeySet
+ * @return the key, if found
+ * @throws NoSuchElementException if the key is not found
  */
-fun keySetOf(vararg keys: Key): KeySet {
-    return KeySet.create(*keys)
-}
+fun KeySet.find(keyName: String): Key = lookup(keyName).orElseThrow()
```
[the remark this answers is not in the document]
Comment (mandoway) [reply src/bindings/jna/libelektra-kotlin/src/main/kotlin/org/libelektra/keySetExt/Util.kt]: Thank you for the suggestions!
I wasn't so sure about the naming either, as it overloads the stdlib function.
It is now more explicit and the get operator is a fine addition.

Implemented all of that. 👍

File: src/bindings/jna/libelektra-kotlin/README.md [hunk 2]
```diff
@@ -189,10 +189,13 @@ Finding Keys:
...
 // Non-nullable, throws NoSuchElementException when not found
-val key: Key = keySet.find("/test/foo")
+val key: Key = keySet.lookupOrThrow("/test/foo")
 
 // Nullable, returns null when not found
-val key: Key? = keySet.findOrNull("/test/foo")
+val key: Key? = keySet.lookupOrNull("/test/foo")
+
+// Get operator, same as lookupOrNull
+val key: Key? = keySet.["/test/foo"]
```
Comment (kodebach) [inline src/bindings/jna/libelektra-kotlin/README.md:198]:
```suggestion
val key: Key? = keySet["/test/foo"]
```

I don't think you need a dot here

**Review by kodebach — APPROVED** [review]

File: src/bindings/jna/libelektra-kotlin/README.md (hunk 2 above)
Comment (mandoway) [reply src/bindings/jna/libelektra-kotlin/README.md]: whoops, typo

**Review by markus2330 — APPROVED** [review]
Great job! Maybe you can have another look at your tutorial (README.md) and also link it from a (to be written) highlight section of _preparation_next_release.md?

**Pull request merged**